### ISSUE TITLE
Conda/Windows CI Failure Mid July 2022

### ISSUE THREAD

**Issue by traversaro** [robotology/robotology-superbuild#1212]
~~~
2022-07-14T02:55:52.6423481Z -- Found JSONCPP: TRUE  
2022-07-14T02:55:52.8728614Z -- Checking for module 'yaml-0.1'
2022-07-14T02:55:52.8862679Z --   No package 'yaml-0.1' found
2022-07-14T02:55:52.8864620Z -- Attempting manual search for yaml
2022-07-14T02:55:52.9046130Z -- Looking for yaml headers - not found
2022-07-14T02:55:52.9046616Z -- Looking for yaml library - not found
2022-07-14T02:55:52.9066240Z CMake Error at C:/Miniconda3/envs/test/Library/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
2022-07-14T02:55:52.9066945Z   Could NOT find YAML (missing: YAML_FOUND)
2022-07-14T02:55:52.9067386Z Call Stack (most recent call first):
2022-07-14T02:55:52.9067942Z   C:/Miniconda3/envs/test/Library/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
2022-07-14T02:55:52.9068550Z   C:/Miniconda3/envs/test/Library/share/cmake/ignition-cmake2/cmake2/FindYAML.cmake:93 (find_package_handle_standard_args)
2022-07-14T02:55:52.9069146Z   C:/Miniconda3/envs/test/Library/lib/cmake/ignition-fuel_tools4/ignition-fuel_tools4-config.cmake:95 (find_package)
2022-07-14T02:55:52.9069747Z   C:/Miniconda3/envs/test/Library/lib/cmake/gazebo/gazebo-config.cmake:241 (find_package)
2022-07-14T02:55:52.9070231Z   cmake/BuildGazeboYARPPlugins.cmake:31 (find_package)
2022-07-14T02:55:52.9070724Z -- Configuring incomplete, errors occurred!
2022-07-14T02:55:52.9075465Z   build/_deps/ycm-src/modules/FindOrBuildPackage.cmake:205 (include)
2022-07-14T02:55:52.9075900Z See also "D:/a/robotology-superbuild/robotology-superbuild/build/CMakeFiles/CMakeOutput.log".
2022-07-14T02:55:52.9076463Z   cmake/RobotologySuperbuildLogic.cmake:19 (find_or_build_package)
2022-07-14T02:55:52.9077957Z See also "D:/a/robotology-superbuild/robotology-superbuild/build/CMakeFiles/CMakeError.log".
2022-07-14T02:55:52.9079242Z   CMakeLists.txt:66 (include)
~~~

**Comment by traversaro** [comment]
So, the first problem is that CMake should not fail if gazebo is not found, but this is a minor problem tracked in https://github.com/osrf/gazebo/issues/3237 .

Then, the problem is understand why yaml is not installed libignition-fuel-tools4 is installed. I noticed that an old libignition-fuel-tools4 (4.0.0) is installed. If I try to force installation of a new fuel-tools, an older ffmpeg is installed, so this is probably a regression related to the ffmpeg44 migration:
~~~
(robsub) C:\src\robotology-superbuild\build2022>mamba install libignition-fuel-tools4=4.4

                  __    __    __    __
                 /  \  /  \  /  \  /  \
                /    \/    \/    \/    \
███████████████/  /██/  /██/  /██/  /████████████████████████
              /  / \   / \   / \   / \  \____
             /  /   \_/   \_/   \_/   \    o \__,
            / _/                       \_____/  `
            |/
        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗
        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗
        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║
        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║
        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║
        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝

        mamba (0.22.1) supported by @QuantStack

        GitHub:  https://github.com/mamba-org/mamba
        Twitter: https://twitter.com/QuantStack

█████████████████████████████████████████████████████████████


Looking for: ['libignition-fuel-tools4=4.4']

conda-forge/win-64                                            No change
conda-forge/noarch                                   8.7MB @   1.1MB/s  8.5s

Pinned packages:
  - python 3.10.*


Transaction

  Prefix: C:\Users\STraversaro\AppData\Local\mambaforge\envs\robsub

  Updating specs:

   - libignition-fuel-tools4=4.4
   - ca-certificates
   - openssl


  Package                    Version  Build             Channel                  Size
---------------------------------------------------------------------------------------
  Install:
---------------------------------------------------------------------------------------

  + yaml                       0.2.5  h8ffe710_2        conda-forge/win-64     Cached

  Change:
---------------------------------------------------------------------------------------

  - gazebo                   11.11.0  h5d48ce7_8        installed
  + gazebo                   11.11.0  hfac46e6_7        conda-forge/win-64       64MB

  Upgrade:
---------------------------------------------------------------------------------------

  - dlfcn-win32                1.2.0  h39d44d4_0        installed
  + dlfcn-win32                1.3.0  h0e60522_0        conda-forge/win-64     Cached
  - jsoncpp                    1.8.4  h1ad3211_1002     installed
  + jsoncpp                    1.9.5  h2d74725_1        conda-forge/win-64     Cached
  - libignition-common3        3.4.0  h21ff451_1        installed
  + libignition-common3       3.13.2  h9bd54ec_2        conda-forge/win-64     Cached
  - libignition-fuel-tools4    4.0.0  hedfdca1_1        installed
  + libignition-fuel-tools4    4.4.0  h3fac903_7        conda-forge/win-64     Cached

  Downgrade:
---------------------------------------------------------------------------------------

  - ffmpeg                     4.4.2  gpl_hc2492c1_105  installed
  + ffmpeg                     4.3.1  ha925a31_0        conda-forge/win-64     Cached

  Summary:

  Install: 1 packages
  Change: 1 packages
  Upgrade: 4 packages
  Downgrade: 1 packages

  Total download: 64MB
~~~

**Comment by traversaro** [comment]
I think I got the problem, see:
~~~
C:\Users\STraversaro>mamba create -n test20220714 libignition-fuel-tools4=4.4 gazebo=11.11.0=*_8

                  __    __    __    __
                 /  \  /  \  /  \  /  \
                /    \/    \/    \/    \
███████████████/  /██/  /██/  /██/  /████████████████████████
              /  / \   / \   / \   / \  \____
             /  /   \_/   \_/   \_/   \    o \__,
            / _/                       \_____/  `
            |/
        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗
        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗
        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║
        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║
        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║
        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝

        mamba (0.22.1) supported by @QuantStack

        GitHub:  https://github.com/mamba-org/mamba
        Twitter: https://twitter.com/QuantStack

█████████████████████████████████████████████████████████████


Looking for: ['libignition-fuel-tools4=4.4', 'gazebo==11.11.0[build=*_8]']

conda-forge/win-64                                          Using cache
conda-forge/noarch                                          Using cache
Encountered problems while solving:
  - package gazebo-11.11.0-h5d48ce7_8 requires libignition-fuel-tools4 >=4.0.0,<4.1.0a0, bd
~~~

Basically gazebo _8 was built with a really old libignition-fuel-tools, probably as the version built with ffmpeg 4.4 was not available, mainly because it seem that the migrator was not aware of the transitive dependency between libignition-fuel-tools4 and gazebo . A new build of gazebo should solve the problem.

**Comment by traversaro** [comment]
This should be fixed by https://github.com/conda-forge/gazebo-feedstock/pull/143 .
